app/src/main/java/io/apicurio/registry/rules/compatibility/AvroCompatibilityChecker.java

```diff
@@ -37,15 +37,15 @@
      * @see CompatibilityChecker#isCompatibleWith(io.apicurio.registry.rules.compatibility.CompatibilityLevel, java.util.List, java.lang.String)
      */
     @Override
-    public boolean isCompatibleWith(CompatibilityLevel compatibilityLevel, List<String> existingSchemaStrings, String proposedSchemaString) {
+    public CompatibilityExecutionResult getIncompatibleDifferences(CompatibilityLevel compatibilityLevel, List<String> existingSchemaStrings, String proposedSchemaString) {
         requireNonNull(compatibilityLevel, "compatibilityLevel MUST NOT be null");
         requireNonNull(existingSchemaStrings, "existingSchemaStrings MUST NOT be null");
         requireNonNull(proposedSchemaString, "proposedSchemaString MUST NOT be null");
 
         SchemaValidator schemaValidator = validatorFor(compatibilityLevel);
 
         if (schemaValidator == null) {
-            return true;
+            return new CompatibilityExecutionResult(true, Collections.emptySet());
```

Comment:
Suggestion:  a static method on the `CompatibilityExecutionResult` called e.g. `empty()` or `compatible()` to provide a shortcut to this.  It could be used in a few places where this is done.

Reply:
done